### ISSUE TITLE
TransferHFile bug

### ISSUE THREAD

**Issue by SteamShon** [kakao/s2graph#93]
in s2loader project, TransferToHFile class use HFileOutputFormat2.configureIncrementalLoad. 

currently it use job.setMapOutputValueClass(classOf[Cell]), which will fail to figure out how to split table(https://github.com/cloudera/hbase/blob/cdh5.3.6-release/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/HFileOutputFormat2.java#L389-L391) 

changing MapOutputValueClass from Cell to KeyValue is necessary for bulk load to work properly.

**Comment by SteamShon** [comment]
Here is what I have tried sofar.

I tried to use hbase-rdd project(https://github.com/unicredit/hbase-rdd).
Then I was not satisfied with the fact that they use string as rowkey, and qualifier. also it expect rdd to be grouped by before. This becomes not feasiable since group by load all adjacent edges on memory, and some data have supernode, which has lots of adjacent edges. 

So I tried to implemented own loader using HFileOutputFormat2.configureIncrementalLoad.
And I found that MR job create too many shuffle writes through spark web UI.

After searching google, found that in hbase master branch, there is hbase-spark component that is integrated from Cloudera(https://github.com/cloudera-labs/SparkOnHBase)

After read through their merged code, thought that it is more optimal with performance, so I decided to use it(anyway, it will be default way to bulk load later). 

Only problem was that only master branch(hbase-2.0.0-SNAPSHOT) has this, and this depends on hbase-common-2.0.0-SNAPSHOT and few other components with 2.0.0. so I brutally copied necessary codes and pasted under loader project.
